lib/simulator/simulatormodule.c

```diff
@@ -519,7 +534,7 @@ static PyObject *next(PyObject *self, PyObject *args)
     PyGILState_STATE gstate;
     gstate = TAKE_GIL();
 
-    if (!PyArg_ParseTuple(args, "l", &hdl)) {
+    if (!PyArg_ParseTuple(args, "O&", gpi_sim_hdl_converter, &hdl)) {
```

Comment:
`hdl` is a `gpi_iterator_hdl` here, not a `gpi_sim_hdl`. They're typedef'd to the same thing, so you can get away with it - but this name is wrong.